### ISSUE TITLE
Let getEvents() find recurring events

### ISSUE THREAD

**Issue by wvrzel** [wvrzel/simpleCalDAV#2]
Right now getEvents() only finds the first one of a recurring event.

I will fix this as soon as I have time.

**Comment by danilphd** [comment]
Hello, thanks for your repo!
Tell me please, are there news about recurring events? I really need this feature.
Or maybe there is a solution without the use of cycles (while, for...)?

**Comment by wvrzel** [comment]
Hi,
I didn't work on this feature since my last post and right now I am laking the time to work for this project.
But I am happy to see, which features are requested. It will be the next thing I will be working on, as soon as I have time for this project again.

I don't think its much effort to get this feature working. Maybe you want to experiment with custom SimpleCalDAVFilters or write xml-filters yourself. Here are three examples of xml-requests to the server. Two of them are for recurring events. Interesting for you is the part within the <C:filter>-tags (See getCustomReport() in SimpleCalDAVClient.php).
http://www.rfcreader.com/#rfc4791_line1623
http://www.rfcreader.com/#rfc4791_line1780
http://www.rfcreader.com/#rfc4791_line1926

I would be happy if you would share your results with us, in case you will be able to solve the problem.

Regards,
Michael

**Comment by rePhat** [comment]
Is this repo ever coming back to life?

**Comment by myzinsky** [comment]
why do you think its dead?

**Comment by rePhat** [comment]
Well I'm looking for some lightweight calDAV client which this one absolutely is. However it seems to have problems with recurring events and the last response is from April.

**Comment by wvrzel** [comment]
*resurrect*

I spent the whole day trying to fix this problem, but it's not that easy.
I am not getting the problem. As soon as an event is recurring, it seems to be always returned from the server (tested with davical 1.1.3.1-1), no matter what the time-range in the filter says.

This seems to contradict the standard. I spent hours, reading about limit-recurrence-set, reading and understanding the examples in the rfc. And I do understand it, but it's not helpful and I'm stuck. I have no idea, what causes this.

Anyone else has an idea?

Well... since no one else contributes to this repo (you're welcome), the live and death of this repo depends on my motivation and the amount of time I am able to offer for this project. Since both are pretty low right now, it's half-dead. I am happy to see, that people are using my software and every post in here is motivating. But reading rfcs, spending serveral days & fixing ugly bugs is no fun.

**Comment by rashnk** [comment]
i tried adding `<C:expand >` filter with the request body, but it returns error, i tested with owncloud server which is SabreDav backend

**Comment by X-Ryl669** [comment]
As I understand the RFC, if the server does not perform the recurrence expansion then it's the client that should do it. The server _should_ return the recurring event if any of the recurrence overlap the timerange (yet, the returned event is not in the timerange). Because the recurrence type is returned, one should handle it by modulo it with the timerange, taking into account all the hard backrules. This is a long work to get right, and it means parsing the ICS data correctly too.

**Comment by rashnk** [comment]
yes, the event ICS contains all the details, dtstart, dtend,weekly, montly, etc. we can do it with a js on client side, i managed to do it with moment-recur.js library, thanks

**Comment by X-Ryl669** [comment]
BTW, Sabredav does expansion (and it's used in many other servers). Davical I'm not sure. So <C:expand > should work (if you want less work on the client side) at least on Baikal, Sabredav, etc...

**Comment by Graschi** [comment]
Hi all

I am using this interface for showing calendar data on meeting room displays with the same problems on recurring events. Can someone explane how to extend the implementation for getting this feature to work?

Any example?

I am glad for getting help
